tests/test_utils.py

```diff
@@ -306,3 +312,480 @@ def check_multistream_classic(fnames):
+            stack.pop(i)
+            for j, _substring in enumerate(substack):
+                stack.insert(i+j, _substring)
+
+    return stack
+
+def read_snotel_swe_obs(filename, science_test_data_dir, items):
+
+    ''' reads in Snotel SWE obs and returns DataFrame
+    '''
+
+    filename_fullpath = os.path.join(science_test_data_dir,
+                                    'datasets',
+                                    items['archive'],
+                                    'observations',
+                                    filename)
+
+    # load snotel obs
+    snotel_swe = pd.read_csv(filename_fullpath,
+                            skiprows=0,
+                            delim_whitespace=True,
+                            names=['YEAR', 'MONTH', 'DAY', 'OUT_SWE'])
+
+    # add datetime object column to snotel DataFrame
+    snotel_swe['DATES'] = pd.to_datetime(snotel_swe.YEAR * 10000 +
+                                        snotel_swe.MONTH * 100 +
+                                        snotel_swe.DAY,
+                                        format='%Y%m%d')
+
+    # remove year, month, day columns of DataFrame
+    snotel_swe.drop(['YEAR', 'MONTH', 'DAY'], inplace=True, axis=1)
+
+    snotel_swe.index = snotel_swe['DATES']
+
+    return(snotel_swe)
+
+def read_vic_42(lat, lng, science_test_data_dir, items):
+    ''' reads output from VIC 4.2
+    '''
+
+    if 'ecflux' in items['compare_to']:
+        vic_42_file = 'en_bal_%s_%s' % (lat, lng)
+        vic_42_dir = os.path.join(science_test_data_dir, 'test_runs',
+                                items['archive'], 'ecflux', 'results')
+
+    elif 'snotel' in items['compare_to']:
+        vic_42_file = 'outfile_%s_%s' % (lat, lng)
+        vic_42_dir = os.path.join(science_test_data_dir, 'test_runs',
+                                items['archive'], 'snotel', 'results')
+
+    else:
+        raise ValueError("this option has not yet been implemented")
+
+    vic_42 = pd.read_csv(os.path.join(vic_42_dir, vic_42_file),
+                            sep='\t',
+                            skiprows=5)
+
+    # remove comment sign from column names in DataFrame
+    vic_42 = vic_42.rename(columns=lambda x: x.replace('#', ''))
+
+    # remove spaces from column names in DataFrame
+    vic_42 = vic_42.rename(columns=lambda x: x.replace(' ', ''))
+
+    # add datetime object column to snotel DataFrame
+    vic_42['DATES'] = pd.to_datetime(vic_42.YEAR * 10000 +
+                                            vic_42.MONTH * 100 +
+                                            vic_42.DAY,
+                                            format='%Y%m%d')
+
+    # remove year, day columns of DataFrame
+    vic_42.drop(['YEAR', 'DAY'], inplace=True, axis=1)
+
+    if ('HOUR' not in vic_42) and ('SEC' in vic_42):
+        # add hour column for groupby
+        vic_42['HOUR'] = (vic_42['SEC'] * (1/3600)
+                                        ).astype(int)
+
+    vic_42.index = vic_42.DATES
```

Comment:
We should make the index a proper datetime index including the subdaily timesteps.